### ISSUE TITLE
App Key Not Generating - Windows/IIS

### ISSUE THREAD

**Issue by kylehemker** [snipe/snipe-it#6249]
#### Expected Behavior (or desired behavior if a feature request)

(what you expect to happen goes here)

`php artisan key:generate` runs and generates app key in .env file.

What's unclear to me is the naming convention of the file, I feel like I'm missing something because I don't see it anywhere in the documentation but I'm also not seeing any reports about it. The only information I could see in the documentation is in regards to editing the config file on Linux (running `cp .env.example .env`) which would imply the file doesn't have a name, and of course Windows won't let you have a file with no name and just an extension. I feel like I'm missing something majorly obvious so apologies if it's right in front of my face in the docs.

-----

#### Actual Behavior

(what actually happens goes here)

Command runs in cmd (in the directory of snipeit) with no output, .env file is not populated with app key

-----

#### Please confirm you have done the following before posting your bug report:

- [X ] I have enabled debug mode 
- [X ] I have read [checked the Common Issues page](https://snipe-it.readme.io/docs/common-issues)

-----
#### Provide answers to these questions:

- Is this a fresh install or an upgrade? 
Fresh
- Version of Snipe-IT you're running
Latest (v4.6.3)
- Version of PHP you're running
5.3.28
- Version of MySQL/MariaDB you're running
Was 10-.3.9 but now 10.0.14 (after seeing reported issues on that version)
- What OS and web server you're running Snipe-IT on
Windows Server 2016/IIS
- What method you used to install Snipe-IT (install.sh, manual installation, docker, etc)
Manual
- WITH DEBUG TURNED ON, if you're getting an error in your browser, include that error
Internal error (500)
- What specific Snipe-IT page you're on, and what specific element you're interacting with to trigger the error
- If a stacktrace is provided in the error, include that too.
- Any errors that appear in your browser's error console.
- Confirm whether the error is reproducible on the demo: https://snipeitapp.com/demo.
- Include any additional information you can find in `storage/logs` and your webserver's logs.
- Include what you've done so far in the installation, and if you got any error messages along the way.
- Indicate whether or not you've manually edited any data directly in the database

Please do not post an issue without answering the related questions above. If you have opened a different issue and already answered these questions, answer them again, once for every ticket. It will be next to impossible for us to help you.

https://snipe-it.readme.io/docs/getting-help

**Comment by kylehemker** [comment]
Doh. That's what I get for not using the command line. /facepalm

**Comment by kylehemker** [comment]
Thanks @snipe, that worked for renaming the file.

Unfortunately I get the same result when running the `key:generate` command in a terminal. It completes with out any message or error but the `.env` file entry for the app key remains the same - `APP_KEY=ChangeMe`.

Thoughts?

**Comment by snipe** [comment]
Try `key:generate --show`. If it gives you a key, copy that into your .env

**Comment by kylehemker** [comment]
No dice, that does not return any output either.

**Comment by snipe** [comment]
Anything in your logs? It's odd that it wouldn't give any output at all.

**Comment by kylehemker** [comment]
I agree, very peculiar. I see a couple parse errors in my PHP logs, though I'm not exactly sure what the issue is.

> [28-Sep-2018 08:40:46 America/Tegucigalpa] PHP Parse error:  syntax error, unexpected T_CLASS, expecting T_STRING or T_VARIABLE or '$' in C:\inetpub\wwwroot\snipe-it\public\index.php on line 55
> [28-Sep-2018 09:01:53 America/Tegucigalpa] PHP Parse error:  syntax error, unexpected T_CLASS, expecting T_STRING or T_VARIABLE or '$' in C:\inetpub\wwwroot\snipe-it\artisan on line 31

So in case those stand out here are the lines from the relevant files.
**C:\inetpub\wwwroot\snipe-it\public\index.php on line 55**
`$kernel = $app->make(Illuminate\Contracts\Http\Kernel::class);`

**C:\inetpub\wwwroot\snipe-it\artisan on line 31**
`$kernel = $app->make(Illuminate\Contracts\Console\Kernel::class);`

I'm not seeing anything in `C:\inetpub\wwwroot\snipe-it\storage\logs` except for a `.gitignore` file and nothing is standing out in the Windows Event logs.

Anywhere else I can look or are the PHP errors telling of anything?

**Comment by kylehemker** [comment]
I was on 5.3 but I upgraded to 7.2.7 prior to reaching out here. It looks like my setup was still using 5.3 though, which has since been remediated.

My key has now been generated!
